### PR TITLE
Openstack new infra tests

### PR DESCRIPTION
**Pull request by rrasouli**
Fixed some of the comments for new tests, new power controls shelve, shelve offload

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit cca555b3d41c4062ad66d46ac772b71246cc8a78:

`cfme/tests/cloud/test_instance_power_control.py`:
- :bangbang: Line 287:13: **E131** *continuation line unaligned for hanging indent*
- :bangbang: Line 292:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 304:13: **E131** *continuation line unaligned for hanging indent*
- :bangbang: Line 311:13: **E131** *continuation line unaligned for hanging indent*
- :bangbang: Line 332:13: **E131** *continuation line unaligned for hanging indent*

`cfme/tests/infrastructure/instance.py`:
- :red_circle: Line 33:7: **N801** *class names should use CapWords convention*
- :red_circle: Line 34:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 37:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 40:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 44:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 47:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 50:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 53:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 56:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 59:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 62:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 66:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 69:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 72:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 75:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 78:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 81:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 84:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 87:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 90:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 93:9: **N805** *first argument of a method should be named 'self'*
- :bangbang: Line 166:1: **E302** *expected 2 blank lines, found 1*

`cfme/tests/infrastructure/test_host_cpu_resources.py`:
- :bangbang: Line 35:17: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 36:21: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 38:17: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 39:21: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 41:17: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 42:21: **E126** *continuation line over-indented for hanging indent*

`cfme/tests/infrastructure/test_host_devices.py`:
- :bangbang: Line 18:1: **E302** *expected 2 blank lines, found 1*

`cfme/tests/infrastructure/test_host_quads_credentials.py`:
- :heavy_exclamation_mark: Line 25:65: **W292** *no newline at end of file*

`cfme/tests/infrastructure/test_host_relationships.py`:
- :bangbang: Line 49:17: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 52:17: **E126** *continuation line over-indented for hanging indent*

`cfme/tests/infrastructure/test_host_security.py`:
- :bangbang: Line 35:17: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 36:21: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 38:17: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 39:21: **E126** *continuation line over-indented for hanging indent*

`cfme/tests/infrastructure/test_host_vm_name.py`:
- :heavy_exclamation_mark: Line 47:1: **W391** *blank line at end of file*

`cfme/tests/infrastructure/test_instance_power_control.py`:
- :bangbang: Line 287:13: **E131** *continuation line unaligned for hanging indent*
- :bangbang: Line 292:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 304:13: **E131** *continuation line unaligned for hanging indent*
- :bangbang: Line 311:13: **E131** *continuation line unaligned for hanging indent*
- :bangbang: Line 332:13: **E131** *continuation line unaligned for hanging indent*

`cfme/tests/infrastructure/test_resources.py`:
- :bangbang: Line 14:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 21:13: **E126** *continuation line over-indented for hanging indent*

`cfme/tests/infrastructure/test_roles_names.py`:
- :bangbang: Line 7:1: **E302** *expected 2 blank lines, found 1*

`cfme/tests/infrastructure/test_status.py`:
- :bangbang: Line 14:1: **E302** *expected 2 blank lines, found 1*

`utils/testgen.py`:
- :bangbang: Line 459:19: **E901** *SyntaxError: invalid syntax*
- :heavy_exclamation_mark: Line 462:1: **W391** *blank line at end of file*

Please, rectify these issues :smirk: .
*CFME QE Bot*

File: cfme/cloud/instance.py [hunk 1]
```diff
@@ -257,6 +262,8 @@ def power_control_from_provider(self, option):
             self.provider.mgmt.resume_vm(self.name)
         elif option == OpenStackInstance.PAUSE:
             self.provider.mgmt.pause_vm(self.name)
+        elif option == OpenStackInstance.shelve_vm(self.name):
```
Comment (jkrocil) [inline cfme/cloud/instance.py]: `== OpenStackInstance.SHELVE`

File: cfme/cloud/instance.py [hunk 2]
```diff
@@ -177,7 +177,12 @@ class OpenStackInstance(Instance):
     STOP = "Stop"
     PAUSE = "Pause"
     RESTART = "Restart"
-
+    # Provider-only shelve options since 5.5
```
Comment (jkrocil) [inline cfme/cloud/instance.py]: What does it mean `provider-only` in this context? If it means its only available on the provider, not in CFME, then it shouldn't be here. Please elaborate.

File: cfme/tests/cloud/test_instance_power_control.py [hunk 3]
```diff
@@ -267,6 +268,74 @@ def test_suspend(
...
+    flash.assert_message_contain("Shelve initiated")
+    testing_instance.wait_for_vm_state_change(
+        desired_state=testing_instance.STATE_SHELVED, timeout=720,
+            from_details=True)
+    soft_assert(
+        provider.mgmt.is_vm_shelved(testing_instance.name),
+        "instance is still running")
+
+
+def test_shelve_offload(
```
Comment (jkrocil) [inline cfme/tests/cloud/test_instance_power_control.py]: This test is missing marks (uncollectif etc).

File: cfme/tests/cloud/test_instance_power_control.py [hunk 4]
```diff
@@ -267,6 +268,74 @@ def test_suspend(
...
+
+    Metadata:
+        test_flag: power_control, provision
+    """
+    testing_instance.wait_for_vm_state_change(
+        desired_state=testing_instance.STATE_ON, timeout=720, from_details=True)
+    check_power_options(soft_assert, testing_instance, 'off')
+    testing_instance.power_control_from_cfme(
+        option=testing_instance.START, cancel=False, from_details=True)
+    flash.assert_message_contain("Pause initiated")
```
Comment (jkrocil) [inline cfme/tests/cloud/test_instance_power_control.py]: We are not initiating `pause`, we are initiating `start`.

File: cfme/tests/cloud/test_instance_power_control.py [hunk 5]
```diff
@@ -267,6 +268,74 @@ def test_suspend(
...
+
+    Metadata:
+        test_flag: power_control, provision
+    """
+    testing_instance.wait_for_vm_state_change(
+        desired_state=testing_instance.STATE_OFF, timeout=720, from_details=True)
+    check_power_options(soft_assert, testing_instance, 'off')
+    testing_instance.power_control_from_cfme(
+        option=testing_instance.START, cancel=False, from_details=True)
+    flash.assert_message_contain("Start initiated")
```
Comment (jkrocil) [inline cfme/tests/cloud/test_instance_power_control.py]: Again, initiating `start`, expected outcome is `STATE_SHELVE_OFFLOAD`...

File: cfme/tests/infrastructure/instance.py [hunk 6]
```diff
@@ -0,0 +1,648 @@
+""" A model of Instances page in CFME."""
```
Comment (jkrocil) [inline cfme/tests/infrastructure/instance.py]: No idea what's going on here. This seems like a copy-paste of `cloud/instance.py` into `infrastructure/instance.py`.

**Comment by mfalesni** [comment]
utils/testgen.py:

:bangbang: Line 459:19: E901 SyntaxError: invalid syntax
:exclamation: Line 462:1: W391 blank line at end of file

^^^ you should rectify this.

You cannot commit a file with a syntax error.

File: utils/testgen.py [hunk 7]
```diff
@@ -522,3 +455,8 @@ def param_check(metafunc, argnames, argvalues):
 
         # apply the mark
         pytest.mark.uncollect()(metafunc.function)
+
+    Contact GitHub API Training Shop Blog About
```
Comment (quarckster) [inline utils/testgen.py]: Why do we need to contact with Github?

File: utils/testgen.py [hunk 8]
```diff
@@ -1,14 +1,11 @@
-"""Test generation helpers
 
+"""Test generation helpers
 Intended to functionalize common tasks when working with the pytest_generate_tests hook.
-
```
Comment (jkrocil) [inline utils/testgen.py]: All the deleted lines in this file seem like a misconfigured IDE to me.

File: utils/testgen.py [hunk 9]
```diff
@@ -522,3 +455,8 @@ def param_check(metafunc, argnames, argvalues):
 
         # apply the mark
         pytest.mark.uncollect()(metafunc.function)
+
+    Contact GitHub API Training Shop Blog About
+
+    © 2016 GitHub, Inc. Terms Privacy Security Status Help
```
Comment (jkrocil) [inline utils/testgen.py]: ???

Comment (mfalesni) [inline utils/testgen.py]: :question:

**Comment by dajoRH** [comment]
Lint report for commit 399eda7af3800d18572c606a83a5c8f991479b07:

`cfme/tests/cloud/test_instance_power_control.py`:
- :bangbang: Line 287:13: **E131** *continuation line unaligned for hanging indent*
- :bangbang: Line 292:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 304:13: **E131** *continuation line unaligned for hanging indent*
- :bangbang: Line 311:13: **E131** *continuation line unaligned for hanging indent*
- :bangbang: Line 332:13: **E131** *continuation line unaligned for hanging indent*

`cfme/tests/infrastructure/instance.py`:
- :red_circle: Line 33:7: **N801** *class names should use CapWords convention*
- :red_circle: Line 34:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 37:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 40:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 44:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 47:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 50:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 53:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 56:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 59:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 62:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 66:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 69:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 72:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 75:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 78:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 81:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 84:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 87:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 90:9: **N805** *first argument of a method should be named 'self'*
- :red_circle: Line 93:9: **N805** *first argument of a method should be named 'self'*
- :bangbang: Line 166:1: **E302** *expected 2 blank lines, found 1*

`cfme/tests/infrastructure/test_host_cpu_resources.py`:
- :bangbang: Line 35:17: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 36:21: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 38:17: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 39:21: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 41:17: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 42:21: **E126** *continuation line over-indented for hanging indent*

`cfme/tests/infrastructure/test_host_devices.py`:
- :bangbang: Line 18:1: **E302** *expected 2 blank lines, found 1*

`cfme/tests/infrastructure/test_host_quads_credentials.py`:
- :heavy_exclamation_mark: Line 25:65: **W292** *no newline at end of file*

`cfme/tests/infrastructure/test_host_relationships.py`:
- :bangbang: Line 49:17: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 52:17: **E126** *continuation line over-indented for hanging indent*

`cfme/tests/infrastructure/test_host_security.py`:
- :bangbang: Line 35:17: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 36:21: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 38:17: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 39:21: **E126** *continuation line over-indented for hanging indent*

`cfme/tests/infrastructure/test_host_vm_name.py`:
- :heavy_exclamation_mark: Line 47:1: **W391** *blank line at end of file*

`cfme/tests/infrastructure/test_instance_power_control.py`:
- :bangbang: Line 287:13: **E131** *continuation line unaligned for hanging indent*
- :bangbang: Line 292:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 304:13: **E131** *continuation line unaligned for hanging indent*
- :bangbang: Line 311:13: **E131** *continuation line unaligned for hanging indent*
- :bangbang: Line 332:13: **E131** *continuation line unaligned for hanging indent*

`cfme/tests/infrastructure/test_resources.py`:
- :bangbang: Line 14:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 21:13: **E126** *continuation line over-indented for hanging indent*

`cfme/tests/infrastructure/test_roles_names.py`:
- :bangbang: Line 7:1: **E302** *expected 2 blank lines, found 1*

`cfme/tests/infrastructure/test_status.py`:
- :bangbang: Line 14:1: **E302** *expected 2 blank lines, found 1*

`utils/testgen.py`:
- :bangbang: Line 459:19: **E901** *SyntaxError: invalid syntax*
- :heavy_exclamation_mark: Line 462:1: **W391** *blank line at end of file*

Please, rectify these issues :smirk: .
*CFME QE Bot*

File: cfme/tests/infrastructure/instance.py (hunk 6 above)
Comment (psav) [inline cfme/tests/infrastructure/instance.py]: Don't we already have ```infrastructure/virtual_machines.py``` to deal with the infra vms?

File: utils/testgen.py (hunk 8 above)
Comment (psav) [inline utils/testgen.py]: Yeh, please don't delete these, they will break docs generation

File: cfme/tests/infrastructure/test_resources.py [hunk 10]
```diff
@@ -0,0 +1,36 @@
+from utils import testgen
+import pytest
+
+
+pytestmark = [pytest.mark.meta(server_roles='+smartproxy +smartstate')]
+
+
+def pytest_generate_tests(metafunc):
```
Comment (psav) [inline cfme/tests/infrastructure/test_resources.py]: I already commented on a nicer way to do this.....

**Pull request closed without merging**